src/android/IALocationPlugin.java

```diff
@@ -668,36 +652,43 @@ private boolean validatePositionArguments(JSONArray args) throws JSONException {
      */
     private void setPosition(final JSONArray args,final CallbackContext callbackContext) throws Exception {
         final IALocation.Builder builder;
-        if (mLocationManager != null){
-            if (validatePositionArguments(args)) {
-                builder = new IALocation.Builder();
-                if (!args.getString(0).trim().equalsIgnoreCase("")) {
-                    builder.withRegion(IARegion.floorPlan(args.getString(0).trim()));
-                }
-                if (args.getJSONArray(1).length() == 2) {
-                    builder.withLatitude(args.getJSONArray(1).getDouble(0));
-                    builder.withLongitude(args.getJSONArray(1).getDouble(1));
-                }
-                cordova.getActivity().runOnUiThread(new Runnable() {
-                    @Override
-                    public void run() {
-                        IALocation iaLocation;
-                        iaLocation = builder.build();
-                        mLocationManager.setLocation(iaLocation);
-                        JSONObject successObject = new JSONObject();
-                        try {
-                            successObject.put("message","Position set");
-                        } catch (JSONException ex) {
-                            Log.e(TAG, ex.toString());
-                            throw new IllegalStateException(ex.getMessage());
-                        }
-                        callbackContext.success(successObject);
-                    }
-                });
+        if (mLocationManager != null) {
+          builder = new IALocation.Builder();
+          String region = args.getString(0).trim();
+          JSONArray location = args.getJSONArray(1);
+          String floorPlanId = args.getString(2).trim();
+          String venueId = args.getString(3).trim();
+
+          if (!venueId.equalsIgnoreCase("")) {
+            builder.withRegion(IARegion.venue(venueId));
+          } else if (!floorPlanId.equalsIgnoreCase("")) {
+            builder.withRegion(IARegion.floorPlan(floorPlanId));
+          } else if (!region.equalsIgnoreCase("")) {
+            builder.withRegion(IARegion.floorPlan(region));
+          } else if (location.length() == 2) {

```

Comment:
Location can be given together with venue / floor plan ID so this should be only `if` (not `else if`)